### PR TITLE
Removing explicit SearchResponse usages in tests - v3

### PR DESCRIPTION
**Pull request by pmpailis**
Tests covered in this PR:

* `org.elasticsearch.percolator.PercolatorQuerySearchIT`

### CODE REVIEW

**Comment by JVerwolf** [comment]
@elasticmachine update branch

File: modules/percolator/src/internalClusterTest/java/org/elasticsearch/percolator/PercolatorQuerySearchIT.java [hunk 1]
```diff
@@ -1064,79 +1151,90 @@ public void testPercolatorQueryViaMultiSearch() throws Exception {
...
+            assertHitCount(item.getResponse(), 2L);
+            assertSearchHits(item.getResponse(), "2", "4");
+            assertThat(item.getFailureMessage(), nullValue());
+
+            item = response.getResponses()[5];
+            assertThat(item.getResponse(), nullValue());
+            assertThat(item.getFailureMessage(), notNullValue());
+            assertThat(item.getFailureMessage(), containsString("[test/6] couldn't be found"));
+        } finally {
+            if (response != null) {
```
Comment (original-brownbear) [inline modules/percolator/src/internalClusterTest/java/org/elasticsearch/percolator/PercolatorQuerySearchIT.java:1230]: Lets back this one out please, this should be resolved by adjusting the ref counting on `MultiSearchResponse` in production so this becomes a single `.decRef()`, not by us doing this manually in tests :)

**Review by original-brownbear — COMMENTED** [review]
Looks good thanks! Lets just back out the multi-search response bit.

File: modules/percolator/src/internalClusterTest/java/org/elasticsearch/percolator/PercolatorQuerySearchIT.java (hunk 1 above)
Comment (pmpailis) [reply modules/percolator/src/internalClusterTest/java/org/elasticsearch/percolator/PercolatorQuerySearchIT.java:1230]: Yeap, makes sense. Was wondering whether we want to eliminate as many test instances as possible in this round and then revisit explicitly `MultiSearchResponse` when we have the `decRef` properly implement. Reverted.

**Comment by pmpailis** [comment]
@elasticmachine update branch

**Review by original-brownbear — COMMENTED** [review]
LGTM

**Pull request merged**